src/main/java/at/ac/tuwien/student/sese2017/xp/hotelmanagement/domain/repository/CustomerSearch.java

```diff
@@ -0,0 +1,58 @@
+package at.ac.tuwien.student.sese2017.xp.hotelmanagement.domain.repository;
+
+import at.ac.tuwien.student.sese2017.xp.hotelmanagement.domain.data.CustomerEntity;
+import java.util.Collections;
+import java.util.List;
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
+import javax.transaction.Transactional;
+import org.apache.lucene.search.Query;
+import org.hibernate.search.jpa.FullTextEntityManager;
+import org.hibernate.search.jpa.Search;
+import org.hibernate.search.query.dsl.QueryBuilder;
+import org.springframework.stereotype.Repository;
+import org.springframework.util.StringUtils;
+
+/**
+ * Provides full text search for CustomerEntity over its name and billingAddress fields.
+ * @author Michael
+ *
+ */
+@Repository
+@Transactional
+public class CustomerSearch {
+
+  @PersistenceContext
+  private EntityManager entityManager;
+
+  /**
+   * Does a full text search over all CustomerEntities for search Strings longer
```

Comment:
What kind of search? A simple matching search or some kind of more complex search?